java.source.base/src/org/netbeans/modules/java/source/indexing/JavaCustomIndexer.java

```diff
@@ -1306,6 +1308,18 @@ private static File dumpHeap(@NonNull final String path) {
 
     private static Pair<Object,Method> heapDumper;
 
+    private static String computeJavacVersion() {
+        if (NoJavacHelper.hasNbJavac()) {
+            File nbJavac = InstalledFileLocator.getDefault().locate("modules/ext/nb-javac-impl.jar", null, false);
+            if (nbJavac != null) {
+                return String.valueOf(nbJavac.lastModified());
```

Comment:
Hmm. Reading something from the JAR, for example `unzip -p ./modules/ext/nb-javac-impl.jar com/sun/tools/javap/resources/version.properties-template` might be better than a time stamp.

Reply:
Well, version.properties-template does not seem to contain any useful information (not really surprising, as it is a template). I doubt the nb-javac jars contain something useable as a version; nb-javac maintainers could add a version info; or we could use the module versions.

But when I was thinking of this:
-to make this work as intended in your original comment, then on every (significant) change to nb-javac the nb-javac maintainers need to change some kind of version (so that the reindex happens) - will they?
-is there a particular usecase where the timestamp would change, the contents of the jar did not and the reindex would be too onerous? All I could imagine are some rare cases during development. Users will (may) install nb-javac, and it will be there with the same timestamp until updated, at which point the timestamp changes and the reindex happens (which is what we wanted, right?)